pyflakes/checker.py

```diff
@@ -1699,7 +1699,34 @@ def CALL(self, node):
                     with self._enter_annotation():
                         self.handleNode(keyword.value, node)
 
-        self.handleChildren(node)
+        if any(_is_typing(node.func, nt, self.scopeStack)
+                for nt in ("TypedDict", "NamedTuple")):
+            with self._enter_annotation(AnnotationState.NONE):
+                self.handleChildren(node)
+
+            attributes = []
+
+            if node.func.id == "TypedDict":
+                if len(node.args) > 1 and isinstance(node.args[1], ast.Dict):
+                    attributes += node.args[1].values
+
+            elif node.func.id == "NamedTuple":
+                # NamedTuple("a", [("a", int)])
+                if len(node.args) > 1 and hasattr(node.args[1], "elts") and all(
```

Comment:
instead of `hasattr(..., elts)` I would prefer a more explicit `and isinstance(node.args[1], (ast.List, ast.Tuple))` -- same for the inner one as well (instead of `getattr)`